src/dos/drive_local.cpp

```diff
@@ -360,10 +360,10 @@ bool localDrive::FindNext(DOS_DTA & dta) {
 	} 
 
 	find_size=(Bit32u) stat_block.st_size;
-	struct tm *time;
-	if ((time=localtime(&stat_block.st_mtime))!=0) {
-		find_date=DOS_PackDate((Bit16u)(time->tm_year+1900),(Bit16u)(time->tm_mon+1),(Bit16u)time->tm_mday);
-		find_time=DOS_PackTime((Bit16u)time->tm_hour,(Bit16u)time->tm_min,(Bit16u)time->tm_sec);
+	struct tm time;
+	if (safe_localtime(&stat_block.st_mtime, &time)) {
+		find_date = DOS_PackDate(time);
+		find_time = DOS_PackTime(time);
```

Comment:
This snippet of code is now comfortably readable instead of inducing eye-glaze due to being a brick wall of whitespace-depleted syntax! Excellent.

This makes the original choice of the "time" variable name feel insufficient given it contains both the date and time:

``` c++
find_date = DOS_PackDate(time);
find_time = DOS_PackTime(time);
```

Given this variable holds both date and time, are variable that involves both would read more correctly.. 
``` c++
find_date = DOS_PackDate(datetime); // or date_time, or dt ..
find_time = DOS_PackTime(datetime); // ... 
```

Reply:
I used simply `datetime` across the board, in all places :)